From a35a052b019a34d9f61169a96366c967b476fc12 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Przemys=C5=82aw=20Buczkowski?= <prem@prem.moe>
Date: Mon, 7 Sep 2026 10:43:45 +0000
Subject: Fix an integer overflow warning on Haiku

ENOMEM is already INT_MIN and therefore negative on Haiku, so we got a warning
failing the tests.

diff --git a/programs/util.c b/programs/util.c
index 065a358..ca8fed6 100644
--- a/programs/util.c
+++ b/programs/util.c
@@ -1162,7 +1162,7 @@ mirrorSrcDir(char* srcDirName, const char* outDirName)
     int status = 0;
     char* newDir = mallocAndJoin2Dir(outDirName, trimPath(srcDirName));
     if (!newDir)
-        return -ENOMEM;
+        return ENOMEM;

     srcMode = getDirMode(srcDirName);
     status = makeDir(newDir, srcMode);
--
2.54.0

